common/app/renderers/DotcomRenderingService.scala

```diff
@@ -65,10 +65,14 @@ class DotcomRenderingService extends GuLogging with ResultWithPreconnectPreload
 
     val start = currentTimeMillis()
 
-    val resp = ws
+    val req = ws
       .url(endpoint)
       .withRequestTimeout(timeout)
       .addHttpHeaders("Content-Type" -> "application/json")
+
+    request.headers.get("X-Request-ID").foreach(requestId => req.addHttpHeaders("X-Request-ID"-> requestId))
```

Comment:
I'm curious why this is a foreach?

Reply:
If you wanted to keep things immutable, you could extract the headers to a `List`, like so:

```scala
val headers: List[(String, String)] = List.concat(
  List("Content-Type" -> "application/json"),
  (request.headers.get("X-Request-ID") match {
    case Some(requestId) => List("X-Request-ID" -> requestId)
    case None            => Nil
  }),
)

val resp = ws
  .url(endpoint)
  .withRequestTimeout(timeout)
  // The “splat operator” `: _*` turns the list into a sequence of individual tuples
  .addHttpHeaders(headers: _*)
  .post(payload)
```